### DISCUSSION TITLE
How to document service arguments?

### DISCUSSION THREAD

**Discussion by tal** [custom-components/pyscript#172] [Q&A] [answered]
The service debug tool will show documentation for a service including the arguments accepted. Is it possible to document the arguments individually so that they display there?

Thanks!

**Comment by craigbarratt** [comment, marked as answer]
Yes, for services the doc string (initial string at the top of the function definition) can contain yaml that defines the arguments.  See the [docs](https://hacs-pyscript.readthedocs.io/en/stable/reference.html#service-service-name).
